R/guide-bins.R

```diff
@@ -298,15 +298,15 @@ GuideBins <- ggproto(
     params$title <- scale$make_title(
       params$title %|W|% scale$name %|W|% title
     )
+    params$key <- key
+    params
+  },
+
+  setup_params = function(params) {
     params$direction <- arg_match0(
-      params$direction %||% direction,
+      params$direction,
       c("horizontal", "vertical"), arg_nm = "direction"
     )
-    if (params$direction == "vertical") {
-      key$.value <- 1 - key$.value
-    }
-
-    params$key <- key
```

Comment:
`guide_bins()` was the only guide that actually did something meaningful upon observing `direction` rather than just validating it.